libtransact/src/contract/context/key_value.rs

```diff
@@ -0,0 +1,1310 @@
+    ///
+    /// # Arguments
+    ///
+    /// * `keys` - A list of natural keys to be deleted from state
+    pub fn delete_state_entries(&self, keys: Vec<K>) -> Result<Vec<String>, KeyValueContextError> {
+        let key_map: HashMap<String, String> = keys
+            .iter()
+            .map(|k| Ok((self.addresser.normalize(k), self.addresser.compute(k)?)))
+            .collect::<Result<HashMap<String, String>, KeyValueContextError>>()?;
+        let state_entry_lists: HashMap<String, StateEntryList> = self.get_state_entry_lists(
+            &key_map
+                .values()
+                .map(ToOwned::to_owned)
+                .collect::<Vec<String>>(),
+        )?;
+
+        let mut deleted_keys = Vec::new();
+        let mut new_entry_lists = Vec::new();
+        let mut delete_lists = Vec::new();
+        key_map.iter().for_each(|(nkey, addr)| {
+            // Fetching the StateEntryList at the corresponding address
+            if let Some(list) = state_entry_lists.get(addr) {
+                // The StateEntry objects will be filtered out of the StateEntryList if it has the
+                // normalized key. This normalized key is added to a list of successfully filtered
+                // entries to be returned.
+                if list.contains(nkey.to_string()) {
+                    let filtered = list
+                        .entries()
+                        .to_vec()
+                        .into_iter()
+                        .filter(|e| e.normalized_key() != nkey)
+                        .collect::<Vec<StateEntry>>();
+                    if filtered.is_empty() {
+                        delete_lists.push(addr.to_string());
+                    } else {
+                        new_entry_lists.push((addr.to_string(), filtered));
+                    }
+                    deleted_keys.push(nkey.to_string());
+                }
+            }
+        });
+        // Delete any StateEntryLists that have an empty list of entries
+        self.context.delete_state_entries(delete_lists.as_slice())?;
+        // Setting the newly filtered StateEntryLists into state using the internal context
+        self.context.set_state_entries(
+            new_entry_lists
+                .iter()
+                .map(|(addr, filtered_list)| {
+                    let new_entry_list = StateEntryListBuilder::new()
+                        .with_state_entries(filtered_list.to_vec())
+                        .build()
+                        .map_err(|err| KeyValueContextError::ProtocolBuildError(Box::new(err)))?;
+                    Ok((addr.to_string(), new_entry_list.into_bytes()?))
+                })
+                .collect::<Result<Vec<(String, Vec<u8>)>, KeyValueContextError>>()?,
+        )?;
+
+        Ok(deleted_keys)
+    }
+
+    /// add_receipt_data adds a blob to the execution result for this transaction.
+    ///
+    /// # Arguments
+    ///
+    /// * `data` - the data to add
```

Comment:
Will "adds a blob" and "the data to add" be clear to the intended audience?  Would it be helpful to add more info or a very brief explanation of why you'd use this?

Reply:
Maybe - "Transaction family-specific data which can be interpreted by application clients."

Reply:
Changed to @vaporos's suggestion

Reply:
@vaporos - I thought the term "transaction family" is deprecated?

Reply:
If not, it should be "transaction-family-specific" (to avoid a possible interpretation as "transaction data that is specific to a family").